tests/integration/test_record.py

```diff
@@ -98,6 +98,169 @@ def record_not_yet_deleted(app):
+        '   <subfield code="u">Inst. Appl. Math., Heidelberg</subfield>'
+        ' </datafield>'
+        ' <datafield tag="980" ind1=" " ind2=" ">'
+        '   <subfield code="a">HEP</subfield>'
+        ' </datafield>'
+        ' <datafield tag="980" ind1=" " ind2=" ">'
+        '   <subfield code="a">THESIS</subfield>'
+        ' </datafield>'
+        ' <datafield tag="980" ind1=" " ind2=" ">'
+        '   <subfield code="a">CORE</subfield>'
+        ' </datafield>'
+        ' <datafield tag="981" ind1=" " ind2=" ">'
+        '   <subfield code="a">222</subfield>'
+        ' </datafield>'
+        '</record>'
+    )
+
+    snippet_deleted = (
+        '<record>'
+        '  <controlfield tag="001">222</controlfield>'
+        '  <controlfield tag="005">20160922232729.0</controlfield>'
+        ' <datafield tag="024" ind1="7" ind2=" ">'
+        '   <subfield code="2">DOI</subfield>'
+        '   <subfield code="a">10.11588/heidok.00021652</subfield>'
+        ' </datafield>'
+        ' <datafield tag="100" ind1=" " ind2=" ">'
+        '   <subfield code="a">Humbert, Pascal</subfield>'
+        '   <subfield code="u">Inst. Appl. Math., Heidelberg</subfield>'
+        ' </datafield>'
+        ' <datafield tag="701" ind1=" " ind2=" ">'
+        '   <subfield code="a">Lindner, Manfred</subfield>'
+        ' </datafield>'
+        ' <datafield tag="856" ind1="4" ind2=" ">'
+        '   <subfield code="u">http://www.ub.uni-heidelberg.de/archiv/21652</subfield>'
+        '   <subfield code="y">U. Heidelberg</subfield>'
+        ' </datafield>'
+        ' <datafield tag="909" ind1="C" ind2="O">'
+        '   <subfield code="o">oai:inspirehep.net:222</subfield>'
+        '   <subfield code="p">INSPIRE:HEP</subfield>'
+        ' </datafield>'
+        ' <datafield tag="980" ind1=" " ind2=" ">'
+        '   <subfield code="a">HEP</subfield>'
+        ' </datafield>'
+        ' <datafield tag="981" ind1=" " ind2=" ">'
+        '   <subfield code="a">222</subfield>'
+        ' </datafield>'
+        '</record>'
+    )
+
+    with app.app_context():
+        json_record_merged = hep.do(create_record(snippet_merged))
+        json_record_merged['$schema'] = 'http://localhost:5000/schemas/records/hep.json'
+
+        json_record_deleted = hep.do(create_record(snippet_deleted))
+        json_record_deleted['$schema'] = 'http://localhost:5000/schemas/records/hep.json'
+
+        with db.session.begin_nested():
+            record_merged = record_upsert(json_record_merged)
+            record_deleted = record_upsert(json_record_deleted)
+            if record_deleted:
+                if record_merged:
+                    r = RecordIndexer()
+                    r.index(record_merged)
+                    r.index(record_deleted)
+                    es.indices.refresh('records-hep')
+        db.session.commit()
+
+    yield
+
+    with app.app_context():
+        _delete_merged_records_from_everywhere('literature', 111, 222)
```

Comment:
Refactoring opportunity: a meta-fixture that takes as argument two JSON records and generates a fixture like this one and the one above. Not urgent, but this repetition stinks a bit.